https/tls_config.go

```diff
@@ -0,0 +1,112 @@
+// Package https allows the implementation of tls
+package https
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/prometheus/common/log"
+	"gopkg.in/yaml.v2"
+)
+
+type Config struct {
+	TLSCertPath string    `yaml:"tlsCertPath"`
+	TLSKeyPath  string    `yaml:"tlsKeyPath"`
+	TLSConfig   TLSStruct `yaml:"tlsConfig"`
+}
+
+type TLSStruct struct {
+	RootCAs            string `yaml:"rootCAs"`
```

Comment:
Do we need this parameter for a TLS server?

Reply:
In case we want to reuse this for [_configuring the Alertmanager cluster traffic_](https://github.com/prometheus/alertmanager/pull/1763/), yes. In that case Alertmanager is both server validating clients as well as client validation servers.

Reply:
Talking about this with @simonpasquier offline. We can remove it here for now and add it again later once we want to reuse it in Alertmanager.